.github/workflows/ci.yaml

```diff
@@ -0,0 +1,95 @@
+name: ci
+on: 
+  push:
+    branches:
+      - master
+  pull_request:
+jobs:
+  test-linux-and-mac:
+    runs-on: ${{ matrix.os }}
+    strategy:
+      matrix:
+        go: ['1.13', '1.14']
+        os: ['ubuntu-latest', 'macos-latest']
```

Comment:
I think Travis CI uses 2 versions on MacOS, Ubuntu 16.04 LTS. 
We should run on Ubuntu latest, and LTS at least. Similarly, latest 2 versions on macOS with latest 2 versions on Xcode.

Reply:
Added 18.04 and 20.04. "ubuntu-latest" is 18.04. There is no support for other versions of macOS besides the latest, unfortunately. See [here](https://help.github.com/en/actions/reference/virtual-environments-for-github-hosted-runners#supported-runners-and-hardware-resources).

I have spent the past two hours trying to mess around the keywords "include" and "exclude" as documented [here](https://help.github.com/en/actions/reference/workflow-syntax-for-github-actions#example-including-additional-values-into-combinations). I am not sure this is a bug or it's the intended behavior, but basically there is no elegant way for me to say I want the additional following two combination: os = 'macos-latest', xcode-version = ['1.13', '1.14']. 

It's pretty clear from this [example](https://help.github.com/en/actions/reference/workflow-syntax-for-github-actions#example-including-additional-values-into-combinations) that you are able to introduce a new variable here. But first of all you are not able to say:
```
include:
  - os: 'macos-latest'
     xcode-version: ['latest', '10.3']
```
because of the syntax issue (i.e. it does not support a list here).

But then you are not able to say
```
include:
  - os: 'macos-latest'
     xcode-version: 'latest'
  - os: 'macos-latest'
     xcode-version: '10.3'
```
either, since the second one (i.e. 10.3) will simply override the first one.

The final outcome is a compromise where I don't use "include" or "exclude" keywords at all and simply let all of them run. This will result 12 runs (see [here](https://github.com/wyk9787/pprof/runs/736723446?check_suite_focus=true)), where 4 of them are repeated (i.e. 2 ubuntu versions * 2 xcode versions). They will still pass without issues but those 4 runs are kind of wasted.

Given the time constraints on Github Action is really not an issue (with 12 runs, they finished in less than 5 minutes), I think this is the best we can do. I do notice Github Action is rapidly developing (e.g. this whole matrix.strategy thing is introduced not too long ago), so maybe there is hope afterall.

Reply:
If I recall correctly, there are either limits on the number of GitHub actions workflows or they're paid; also, more tests creates more opportunity for flakiness, and the listing of tests run might be somewhat confusing. 

It might be worth some repetition (specifically, splitting out Mac and Linux test cases) to avoid this.